# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLMemoryTracker.h
# Commit: 7e2a2b1b416f
# Full Hash: 7e2a2b1b416f1a64c6548aab29e92acc9554ae7d
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-01-09 03:40:18
# Regressor Bug: 1477756
# File Overlap Count: 1
# Description:
#   Bug 1477756 - Fix all webgl regression tests according to CI. r=handyman
#   
#   (This is a combination of 31 commits)
#   
#   * Fix Linux compilation.
# ==============================================================================

diff -r ccfa767dba64 -r 7e2a2b1b416f dom/canvas/WebGLMemoryTracker.h
--- a/dom/canvas/WebGLMemoryTracker.h	Wed Jan 08 22:19:16 2020 +0000
+++ b/dom/canvas/WebGLMemoryTracker.h	Wed Jan 08 22:19:23 2020 +0000
@@ -6,68 +6,17 @@
 #ifndef WEBGL_MEMORY_TRACKER_H_
 #define WEBGL_MEMORY_TRACKER_H_
 
-#include "mozilla/StaticPtr.h"
 #include "nsIMemoryReporter.h"
-#include <unordered_set>
 
 namespace mozilla {
 
-class HostWebGLContext;
-
 class WebGLMemoryTracker : public nsIMemoryReporter {
+ public:
   NS_DECL_THREADSAFE_ISUPPORTS
   NS_DECL_NSIMEMORYREPORTER
 
-  static StaticRefPtr<WebGLMemoryTracker> sUniqueInstance;
-
-  static RefPtr<WebGLMemoryTracker> Create();
-
-  static RefPtr<WebGLMemoryTracker> Get() {
-    if (!sUniqueInstance) {
-      sUniqueInstance = WebGLMemoryTracker::Create().get();
-    }
-    return sUniqueInstance;
-  }
-
-  // Here we store plain pointers, not RefPtrs: we don't want the
-  // WebGLMemoryTracker unique instance to keep alive all
-  // WebGLContexts ever created.
-  std::unordered_set<const HostWebGLContext*> mContexts;
-
- public:
-  static void AddContext(const HostWebGLContext* c) {
-    const auto tracker = Get();
-    tracker->mContexts.insert(c);
-  }
-
-  static void RemoveContext(const HostWebGLContext* c) {
-    const auto tracker = Get();
-    tracker->mContexts.erase(c);
-  }
-
  private:
-  WebGLMemoryTracker();
-  virtual ~WebGLMemoryTracker();
-
-  static int64_t GetTextureMemoryUsed();
-
-  static int64_t GetTextureCount();
-
-  static int64_t GetBufferMemoryUsed();
-
-  static int64_t GetBufferCacheMemoryUsed();
-
-  static int64_t GetBufferCount();
-
-  static int64_t GetRenderbufferMemoryUsed();
-
-  static int64_t GetRenderbufferCount();
-
-  static int64_t GetShaderSize();
-
-  static int64_t GetShaderCount();
-
-  static int64_t GetContextCount() { return Get()->mContexts.size(); }
+  virtual ~WebGLMemoryTracker() = default;
 };
 
 }  // namespace mozilla